File: backend/app/core/analytics.py
"""
Émission d'événements produit vers PostHog — funnel démo → abonnement Tawla
(tooling business interne, pas une fonctionnalité restaurateur/convive).
Appelé depuis subscription_payments.py pour purchase_completed. Dégradation
gracieuse : sans POSTHOG_API_KEY (voir .env.example), aucun envoi n'est
tenté — jamais bloquant pour le flux de paiement.
"""
from posthog import Posthog

from app.core.config import settings

_client = Posthog(settings.posthog_api_key, host=settings.posthog_host) if settings.posthog_api_key else None


def capture_event(distinct_id: str, event: str, **properties) -> None:
    if not _client:
        return
    _client.capture(event, distinct_id=distinct_id, properties=properties)

File: frontend/lib/analytics.ts
import posthog from "posthog-js";

const POSTHOG_KEY = process.env.NEXT_PUBLIC_POSTHOG_KEY;
const POSTHOG_HOST = process.env.NEXT_PUBLIC_POSTHOG_HOST || "https://eu.i.posthog.com";

let initialized = false;

/**
 * Analytics produit (PostHog) — funnel démo → abonnement Tawla, pas une
 * fonctionnalité restaurateur/convive. Appelé une seule fois par
 * <Analytics /> au montage du layout racine. Dégradation gracieuse : sans
 * NEXT_PUBLIC_POSTHOG_KEY (voir .env.local.example), reste non initialisé —
 * trackEvent/trackPageview deviennent des no-op.
 */
export function initAnalytics() {
  if (initialized || !POSTHOG_KEY) return;
  posthog.init(POSTHOG_KEY, {
    api_host: POSTHOG_HOST,
    capture_pageview: false,
    person_profiles: "identified_only",
  });
  initialized = true;
}

export function trackEvent(event: string, properties?: Record<string, unknown>) {
  if (!initialized) return;
  posthog.capture(event, properties);
}

export function trackPageview(url: string) {
  if (!initialized) return;
  posthog.capture("$pageview", { $current_url: url });
}
